### DISCUSSION TITLE
Sample correction for quantile of nonconformity scores?

### DISCUSSION THREAD

**Discussion by pat-alt** [JuliaTrustworthyAI/ConformalPrediction.jl#17] [Q&A]
This may be a really silly one ... 😅 

In the literature the formula for the $(1-\alpha)$ quantile is typically explicitly stated with sample correction: the $(1-\alpha)$ quantile is defined as the $\frac{\lceil (1 - \alpha) (n+1)\rceil}{n}$-th smallest value of the empirical distribution of non-conformity scores (see, for example, [here](https://arxiv.org/pdf/2107.07511.pdf)). [Barber et al. (2020)](https://arxiv.org/abs/1905.02928) denote this $(1-\alpha)$ quantile as $\hat{q}_{n,\alpha}^+\\{v\\}$.

In the package I therefore initially made this explicit (see [here](https://github.com/pat-alt/ConformalPrediction.jl/blob/5f7f082d2eb69d4242b38d9538cd2c035150ce39/src/ConformalModels/conformal_models.jl#L55)) like so,

```julia
"""
    qplus(v::AbstractArray, conf_model::ConformalModel)
Computes the empirical quantile `q̂` of the calibrated conformal scores for a user chosen coverage rate `(1-α)`.
"""
function qplus(v::AbstractArray, conf_model::ConformalModel)
    n = length(v)
    p̂ = ceil(((n+1) * conf_model.coverage)) / n
    p̂ = clamp(p̂, 0.0, 1.0)
    q̂ = Statistics.quantile(v, p̂)
    return q̂
end
```

where `v` corresponds to the vector of nonconformity scores. 

But I noted that when I implement the quantile in this way, it doesn't hold that $\hat{q}^{-}(v) =- \hat{q}^{+}(-v)$ where $\hat{q}^{-}(v)$ denotes the $\alpha$ quantile. I checked how this is implemented in MAPIE and it turns out they do not explicitly implement the correction (see [here](https://mapie.readthedocs.io/en/latest/_modules/mapie/regression.html#MapieRegressor)). 

I assume the sample correction is just done under the hood by `Statistics.quantile` and that my initial approach was therefore wrong? Since #18 I'm following the same approach as in MAPIE, just applying `Statistics.quantile` directly to `v` for a given coverage rate $(1-\alpha)$.

Any thoughts/comments would be much appreciated!

**Comment by pat-alt** [comment]
@LacombeLouis @gmartinonQM perhaps you can help? Great work on MAPIE by the way, awesome stuff!

**Reply by gmartinonQM** [reply]
The finite sample correction can only "displace" the quantile by one row in your array, so we played with the option `method="higher/lower"` in numpy : https://numpy.org/doc/stable/reference/generated/numpy.nanquantile.html.

**Reply by pat-alt** [reply]
Ah that makes sense, cheers!
